EmotiBitNvmController.h

```diff
@@ -0,0 +1,216 @@
+/**************************************************************************/
+/*!
+	@file     EmotiBitNvmController.h
+	This is a library to handle Read/Write Operations to NVM on EmotiBit.
+	EmotiBit invests time and resources providing this open source code,
+	please support EmotiBit and open-source hardware by purchasing
+	products from EmotiBit!
+	Written by Nitin Nair for EmotiBit.
+	BSD license, all text here must be included in any redistribution
+*/
+/**************************************************************************/
+#ifndef EMOTIBIT_MEMORY_CONTROLLER_H
+#define EMOTIBIT_MEMORY_CONTROLLER_H
+
+//#include "EmotiBitVersionController.h"
+#include "EmotiBit_Si7013.h"
+#include "SparkFun_External_EEPROM.h"
+
+#define EMOTIBIT_EEPROM_I2C_ADDRESS 0x50
+
+// Uncomment the macro to enable serial prints.
+//#define DEBUG_SERIAL
+
+class EmotiBitNvmController
+{
+public:
+	
+	enum class NvmType {
+		UNKNOWN = -1,
+		OTP = 0,
+		EEPROM = 1,
+		length
+	};
+	
+	struct ConstEepromAddr
+	{
+		static const size_t NUM_MAP_ENTRIES = 0;
+		static const size_t MEMORY_MAP_BASE = 1;
+	};
+	
+	enum class DataType
+	{
+		ENTIRE_NVM = -1,
+		VARIANT_INFO = 0,
+		EDA = 1,
+		length
+	};
+	struct EepromMemoryMap
+	{
+		uint32_t address = 0;
+		uint32_t dataSize = 0;
+	}map[(uint8_t)DataType::length];
+
+	struct Si7013OtpMemoryMap
+	{
+		static const uint8_t EDL_DATA_START_ADDR = 0x82;
+		static const uint8_t EDL_DATA_SIZE = 20; // 5 floats
+		static const uint8_t EDR_DATA_START_ADDR = 0xB0;
+		static const uint8_t EDR_DATA_SIZE = 4;  // 1 float
+		static const uint8_t EMOTIBIT_VERSION_ADDR = 0xB7;
+		static const uint8_t DATATYPE_VERSION_ADDR = 0xB6;
+		static const uint8_t OTP_SIZE_BYTES = 54; 
+	};
+
+	struct EmotiBitEepromSettings
+	{
+		size_t capacityBytes;
+		size_t pageSizeBytes;
+	}emotibitEepromSettings;
+
+	enum class Status
+	{
+		SUCCESS = 0,
+		NVM_TYPE_UNKNOWN = 1,
+		MEMORY_NOT_UPDATED = 2,
+		OUT_OF_BOUNDS_ACCESS = 3,
+		I2C_WRITE_ERROR = 4,
+		CONTROLLER_BUSY = 5,
+		INVALID_DATA_TO_WRITE = 6,
+		NVM_TYPE_NOT_SUPPORTED = 7,
+		OTHER_FAILURE
+	};
+
+	struct Buffer
+	{
+		uint8_t* data = nullptr;
+		uint32_t dataSize = 0;
+		uint8_t datatypeVersion = 0;
+		DataType datatype = DataType::length;
+
+		/*!
+			@brief Sets the datatype of the calling buffer.
+			@param datatype Type of data being stored in the buffer. 
+		*/
+		void setDatatype(DataType datatype);
```

Comment:
I think it's a matter of semantics, but struct is basically a class, with everything public by default.
But i guess since i did add methods, i can change it to a `class`